src/ua.c

```diff
@@ -1982,6 +2021,59 @@ struct sipevent_sock *uag_sipevent_sock(void)
 }
 
 
+static bool uri_match_transport(const struct uri *accu, enum sip_transp tp)
+{
+	struct pl pl;
+	int err;
+
+	err = msg_param_decode(&accu->params, "transport", &pl);
+	if (err)
+		return true;
+
+	return !pl_strcasecmp(&pl, sip_transp_name(tp));
+}
+
+
+static bool uri_match_af(const struct uri *accu, const struct uri *peeru)
+{
+#ifdef HAVE_INET6
+	struct sa sa1;
+	struct sa sa2;
+	int err;
+#endif
+
+	/* we list cases where we know there is a mismatch in af */
+#ifdef HAVE_INET6
+	if (accu->af == AF_INET && peeru->af == AF_INET6)
+		return false;
+
+	if (accu->af == AF_INET6 && peeru->af == AF_INET)
+		return false;
```

Comment:
perhaps these 5 lines could be replaced by:

```c
if (accu->af != peer->af)
    return false;
```

Reply:
If I add another exception for accu->af == AF_UNSPEC. I'll write the function differently.